### PR TITLE
RESTful Routing: clarify usage of id option with $routes->resources

### PR DESCRIPTION
**Pull request by mimo74**
Added a Note to the example for the usage of $routes->resources as doing this wrong may lead to time consuming and frustrating error search

### CODE REVIEW

File: en/development/routing.rst [hunk 1]
```diff
@@ -1034,6 +1034,10 @@ PATCH       /recipes/123.format   RecipesController::edit(123)
 DELETE      /recipes/123.format   RecipesController::delete(123)
 =========== ===================== ==============================
 
+.. note::
+
+the default for $routes->resources requires your IDs to match integer values or UUIDs. If your IDs do not match that you will have to supply the regular expression fragment to use when matching IDs in the additional id option: $builder->resources('Recipes', ['id' => '.*']). Otherwise your routes will not work.
```
Comment (markstory) [inline en/development/routing.rst]:
```suggestion
    The default for pattern for resource IDs only matches integers or UUIDs. If your IDs are different you will have to supply a regular expression pattern via the  ``id`` option. E.g. ``$builder->resources('Recipes', ['id' => '.*'])``. 
```

Comment (dereuromark) [inline en/development/routing.rst]: You should code fence the code pieces here.

File: en/development/routing.rst [hunk 2]
```diff
@@ -1036,7 +1036,7 @@ DELETE      /recipes/123.format   RecipesController::delete(123)
 
 .. note::
 
-the default for $routes->resources requires your IDs to match integer values or UUIDs. If your IDs do not match that you will have to supply the regular expression fragment to use when matching IDs in the additional id option: $builder->resources('Recipes', ['id' => '.*']). Otherwise your routes will not work.
+    The default for pattern for resource IDs only matches integers or UUIDs. If your IDs are different you will have to supply a regular expression pattern via the  ``id`` option. E.g. ``$builder->resources('Recipes', ['id' => '.*'])``.
```
Comment (jadb) [inline en/development/routing.rst]: Wrap the text at ~80 chars / line?

```suggestion
    The default for pattern for resource IDs only matches integers or UUIDs. 
    If your IDs are different you will have to supply a regular expression pattern 
    via the  ``id`` option. E.g. ``$builder->resources('Recipes', ['id' => '.*'])``. 
```

**Comment by markstory** [comment]
Thank you 👍

**Pull request merged**